src/options/set.rs

```diff
@@ -81,10 +81,13 @@ pub fn set_options(
     // Set features
     let mut builtin_features = features::make_builtin_features();
 
-    // --color-only is used for interactive.diffFilter (git add -p) and side-by-side cannot be used
+    // --color-only is used for interactive.diffFilter (git add -p) and
+    // side-by-side, file_style, and hunk_header_style cannot be used
     // there (does not emit lines in 1-1 correspondence with raw git output). See #274.
     if config::user_supplied_option("color-only", arg_matches) {
         builtin_features.remove("side-by-side");
+        builtin_features.remove("file-style");
+        builtin_features.remove("hunk-header-style");
```

Comment:
Have you confirmed that these changes are necessary? I don't think they have any effect, since the keys in `builtin_features` hash map should just be the [builtin features](https://github.com/dandavison/delta/tree/master/src/features), i.e. `side-by-side`, `line-numbers`, `navigate`, `color-only`, `diff-so-fancy`, `diff-highlight`, `hyperlinks`.